server/pbench/bin/pbench-server-prep-version-001

```diff
@@ -0,0 +1,174 @@
+# load common things
+opts=$SHELLOPTS
+case $opts in
+    *xtrace*)
+        dir=$(dirname $(which $0))
+        PROG=$(basename $(which $0))
+        ;;
+    *)
+        dir=$(dirname $0)
+        PROG=$(basename $0)
+        ;;
+esac
+
+export CONFIG=/opt/pbench-server/lib/config/pbench-server.cfg
+
+. $dir/pbench-base.sh
+###########################################################################
+
+log_init $(basename $0) $LOGSDIR/$(basename $0)
+
+tmp=$TMP/$PROG.$$
+
+trap 'rm -rf $tmp' EXIT INT QUIT
+
+mkdir -p $tmp
+list=$tmp/list.check
+status=$tmp/status
+> $status
+results_dir=$(getconf.py receive-dir pbench-server)
+find ${results_dir} -maxdepth 2 -name *.tar.xz.check > $list
+
+archive=$(getconf.py archive-dir pbench-server)
+quarantine=$(getconf.py quarantine-dir pbench-server)
+duplicates=$(getconf.py duplicates-dir pbench-server)
+
+typeset -i ntbs=0
+typeset -i nerrs=0
+typeset -i ndups=0
+
+echo $TS
+while read ck ;do
+    # full pathname
+    tb=${ck%.check}
+
+    # directory
+    tbdir=$(dirname ${tb})
+
+    # resultname: get the basename foo.tar.xz and then strip the .tar.xz
+    resultname=$(basename ${tb})
+    resultname=${resultname%.tar.xz}
+
+    # hostname is the last component of the directory part of the full path
+    hostname=$(basename ${tbdir})
+
+    dest=$archive/$hostname
+    # if there is a prefix file with the old name, normalize it now
+    prefixname=${tbdir}/prefix.${resultname}
+    if [ -f ${prefixname} ] ;then
+        mv ${prefixname} ${tbdir}/${resultname}.prefix
+    fi
+
+    # sanity checks
+    typeset -i err=0
+    if [ ! -f ${tb} ] ;then
+        echo "$TS: {TB} file does not exist" >&4
+        err=1
+    fi
+    if [ ! -f ${tb}.md5 ] ;then
+        echo "$TS: {TB}.md5 file does not exist" >&4
+        err=1
+    fi
+    typeset -i dup=0
+    if [ -f ${dest}/${resultname}.tar.xz ] ;then
+        dup=1
+    fi
+    if [ -f ${dest}/${resultname}.tar.xz.md5 ] ;then
+        dup=1
+    fi
+
+    if [ ${err} -eq 1 ] ;then
+        # quarantine the tarball
+        mkdir -p ${quarantine}/${hostname}
+        sts=$?
+        if [ $sts != 0 ] ;then
+            echo "$TS: Failed: mkdir ${quarantine}/${hostname} with status $sts" >&4
+            continue
+        fi
+        dest=${quarantine}/${hostname}
+        nerrs=$nerrs+1
+        echo "$TS: quarantined ${tb} moved to ${quarantine}/${hostname}" >> $status
+    elif [ ${dup} -eq 1 ] ;then
+        mkdir -p ${duplicates}/${hostname}
+        sts=$?
+        if [ $sts != 0 ] ;then
+            echo "$TS: Failed: mkdir ${duplicates}/${hostname} with status $sts" >&4
+            continue
+        fi
+        dest=${duplicates}/${hostname}
+        ndups=$ndups+1
+        echo "$TS: duplicate ${tb} moved to $dest" >> $status
+    fi
+
+    prefix=${tb%.tar.xz}.prefix
+    if [ -f ${prefix} ] ;then
+        mkdir -p ${dest}/.prefix
+        mv ${prefix} ${dest}/.prefix
+    fi
+    mv ${tb} ${tb}.md5 $dest
```

Comment:
if the move fails we should not remove the check file so that it can be picked up in the next iteration.
`sts=$?
    if [ $sts != 0 ] ;then
        echo "$TS: Failed: mv ${tb} ${tb}.md5 ${dest} with status $sts" >&4
        continue
    fi
`

Reply:
Not sure about that: it depends on why the `mv` failed. If it has a chance of succeeding next time, then we might do what you suggest. But I don't want it to try forever if there is no chance of succes. I'd rather move things aside and let a human know. In fact, we might just leave the tarball and the md5 there and delete the check file before notifying. A human can then figure out what's wrong, fix it, and then `touch <result>.tar.xz.check` and let the script retry it.